nucliadb/nucliadb/export_import/consumers.py

```diff
@@ -0,0 +1,270 @@
+import asyncio
+
+import nats
+from nats.aio.client import Msg
+from nucliadb_protos.resources_pb2 import CloudFile
+
+from nucliadb.common.context import ApplicationContext
+from nucliadb.export_import import logger
+from nucliadb.export_import.datamanager import ExportImportDataManager
+from nucliadb.export_import.exceptions import MetadataNotFound
+from nucliadb.export_import.exporter import (
+    export_entities,
+    export_labels,
+    export_resource,
+)
+from nucliadb.export_import.importer import IteratorExportStream, import_kb
+from nucliadb.export_import.models import (
+    ExportMessage,
+    ExportMetadata,
+    ImportMessage,
+    ImportMetadata,
+    Status,
+)
+from nucliadb.export_import.utils import get_broker_message, iter_kb_resource_uuids
+from nucliadb_telemetry import errors
+from nucliadb_utils import const
+from nucliadb_utils.nats import MessageProgressUpdater
+from nucliadb_utils.settings import nats_consumer_settings
+from nucliadb_utils.storages.storage import StorageField
+
+UNRECOVERABLE_ERRORS = (MetadataNotFound,)
+
+EXPORT_MAX_RETRIES = 5
+
+KB_EXPORTS = "exports/{export_id}"
+KB_IMPORTS = "imports/{import_id}"
+
+
+class BaseConsumer:
+    stream: const.Streams
+
+    def __init__(
+        self,
+        context: ApplicationContext,
+        partition: int,
```

Comment:
Unsure yet what to do about partitioning on export/import streams.

Reply:
Do not do any partitioning. We don't need to worry about ordering so no partitioning necessary